Frontends/glslang/GlslangToTopVisitor.cpp

```diff
@@ -1391,7 +1391,12 @@ bool TGlslangToTopTraverser::visitAggregate(glslang::TVisit visit, glslang::TInt
         result = createUnaryIntrinsic(node->getOp(), precision, operands.front());
         break;
     default:
-        result = createIntrinsic(node->getOp(), precision, uniform, operands, glslangOperands.front()->getAsTyped()->getBasicType() == glslang::EbtUint);
+        // Check first for intrinsics that can be done natively
+        if (node->getOp() == glslang::EOpMix && 
+              gla::GetBasicTypeID(operands.front()) == llvm::Type::IntegerTyID)
+            result = llvmBuilder.CreateSelect(operands[2], operands[1], operands[0]);
+        else
+            result = createIntrinsic(node->getOp(), precision, uniform, operands, glslangOperands.front()->getAsTyped()->getBasicType() == glslang::EbtUint);
```

Comment:
As you suggested offline, maybe we can move ahead with this design for the moment. I think it is elegant how all "regular" builtins filter down to this point and their operands are processed rather simply. Once we have more cases we can design something to deal with the operand order a little more cleanly.